Validator/FormatValidator.php

```diff
@@ -50,8 +50,16 @@ public function validate($value, Constraint $constraint)
             }
         }
 
-        if (!array_key_exists($value->getDefaultFormat(), $formats)) {
-            $this->context->addViolation('invalid format');
+        $galleryDefaultFormat = $value->getDefaultFormat();
+
+        if ($formats) {
+            if (!array_key_exists($galleryDefaultFormat, $formats) && $galleryDefaultFormat !== 'reference') {
+                $this->context->addViolation('invalid format');
+            }
+        } else {
+            if ($galleryDefaultFormat !== 'reference') {
```

Comment:
No. I think here `if` is more clear because the flow has 2 difference path depending on the value of `format`.